includes/payments/class-payments-query.php

```diff
@@ -746,24 +746,33 @@ private function remap_args() {
 			$arguments['status'] = edd_get_payment_status_keys();
 		}
 
-		if ( isset( $this->args['meta_query'] ) && is_array( $this->args['meta_query'] ) ) {
-			foreach ( $this->args['meta_query'] as $meta ) {
+		if ( isset( $arguments['meta_query'] ) && is_array( $arguments['meta_query'] ) ) {
+			foreach ( $arguments['meta_query'] as $meta_index => $meta ) {
 				if ( ! empty( $meta['key'] ) ) {
 					switch ( $meta['key'] ) {
 						case '_edd_payment_customer_id':
 							$arguments['customer_id'] = absint( $meta['value'] );
+							unset( $arguments['meta_query'][ $meta_index ] );
```

Comment:
Small piece of general feedback. This code will work, but I would caution against it.

`$arguments` is being iterated through inside of a `foreach` loop. In the middle of that loop, its current index is being unset. If in the future, this argument index needs to be referred to lower (outside the `switch` but still inside the `foreach` loop) it will be gone and trigger a notice. Also, if PHP changes the way Iterators work (clone vs. byref) this may trigger a warning/error, unshifting the original array, or something else weird.

Instead of unsetting mid-iteration, it may be safer to do the opposite, and only add the indexes you want to a new array, and then replace `$arguments` with that new array after the `foreach` has completed.

Reply:
Thank you @JJJ . In the interest of getting beta-2 out, I'm going to merge this for now but will be opening another issue to address refactoring this.